packages/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ImagePickerUtils.kt

```diff
@@ -1,92 +1,149 @@
+    Log.w(TAG, "Image type not supported. Falling back to JPEG instead.")
+    ".jpeg"
   }
-  else -> ".jpg"
+  else -> ".jpeg"
 }
 
-class ExifDataHandler(private val uri: Uri) {
-  fun copyExifData(newUri: Uri, contentResolver: ContentResolver) {
-    if (uri == newUri) {
-      return
+internal fun String.toBitmapCompressFormat(): Bitmap.CompressFormat = when {
+  this.contains("png") ||
+  this.contains("gif") ||
+  this.contains("bmp") -> {
+    // The result image won't ever be a GIF of a BMP as the cropper doesn't support it.
+    Bitmap.CompressFormat.PNG
+  }
+  else -> {
+    if (!this.contains("jpeg")) {
+      Log.w(TAG, "Image type not supported. Falling back to JPEG instead.")
     }
-    contentResolver.openInputStream(uri)?.use { input ->
-      val oldExif = ExifInterface(input)
-      newUri.path?.let {
-        val newExif = ExifInterface(it)
-        for ((_, exifTag) in ImagePickerConstants.exifTags) {
-          val value = oldExif.getAttribute(exifTag)
-          if (value != null &&
-            exifTag != ExifInterface.TAG_IMAGE_LENGTH &&
-            exifTag != ExifInterface.TAG_IMAGE_WIDTH &&
-            exifTag != ExifInterface.TAG_PIXEL_X_DIMENSION &&
-            exifTag != ExifInterface.TAG_PIXEL_Y_DIMENSION &&
-            exifTag != ExifInterface.TAG_ORIENTATION
-          ) {
-            newExif.setAttribute(exifTag, value)
-          }
-        }
-        try {
-          newExif.saveAttributes()
-        } catch (e: IOException) {
-          Log.w(TAG, "Couldn't save Exif data: ${e.message}", e)
-        }
+    Bitmap.CompressFormat.JPEG
+  }
+}
+
+internal fun MediaMetadataRetriever.extractInt(key: Int): Int {
+  return this.extractMetadata(key)?.toInt() ?: throw FailedToExtractVideoMetadataException()
+}
+
+/**
+ * Copy the media file from `sourceUri` to `destinationUri`.
+ *
+ * @param sourceUri uri to the file to copy the data from
+ * @param targetFile file to save the media data into
+ */
+internal suspend fun copyFile(
+  sourceUri: Uri,
+  targetFile: File,
+  contextProvider: ContextProvider,
+) = runInterruptible {
+  val destinationUri = Uri.fromFile(targetFile)
+
+  // source and destination are the same file
+  if (sourceUri.compareTo(destinationUri) == 0) {
+    return@runInterruptible
+  }
+
+  try {
+    contextProvider.context.contentResolver.openInputStream(sourceUri)?.use { inputStream ->
+      FileOutputStream(targetFile).use { fileOutputStream ->
+        inputStream.copyTo(fileOutputStream)
+        return@runInterruptible
       }
+    } ?: throw FailedToReadFileException(sourceUri.toFile())
+  } catch (cause: FileNotFoundException) {
+    throw FailedToWriteFileException(targetFile, cause)
+  }
+}
+
+/**
+ * The same as [CoroutineScope.launch] but with built-in exception handler
+ * that relays caught exceptions to [Promise.reject] ensuring that these are wrapped in [CodedException].
+ *
+ * TODO(@bbarthec): inspect whether not to add this to the [AsyncFunction] definition
```

Comment:
It was already built into the `e-m-c`.

If you want to transform the function's body into the `coroutine`, you need to use a construction like this:
```
AsyncFunction("<name>")  Coroutine { <body> }
```
It should also handle all uncaught exceptions.

Reply:
And also, if you want to dispatch some coroutines, you can use a `coroutineScope` from the module class or if you're in the `AsyncFunction ... Coroutine` block, you will have access to the function scope there.